Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -1720,14 +1720,20 @@ def _handle_ExprNode(self, node, do_visit_children):
                 cname = EncodedString(Naming.genexpr_arg_prefix + Symtab.punycodify_name(str(name_source)))
                 name_decl = Nodes.CNameDeclaratorNode(pos=pos, name=name)
                 type = node.type
-                if type.is_reference and not type.is_fake_reference:
-                    # It isn't obvious whether the right thing to do would be to capture by reference or by
-                    # value (C++ itself doesn't know either for lambda functions and forces a choice).
-                    # However, capture by reference involves converting to FakeReference which would require
-                    # re-analysing AttributeNodes. Therefore I've picked capture-by-value out of convenience
-                    # TODO - could probably be optimized by making the arg a reference but the closure not
-                    # (see https://github.com/cython/cython/issues/2468)
-                    type = type.ref_base_type
+                while ((type.is_reference and not type.is_fake_reference) or
+                        type.is_cv_qualified):
```

Comment:
BTW, this looks so generic (unpack the plain type) that it could be a helper function. From a quick look, there are a few places where we unpack both `const` and reference types.

Since this is the first time we use a loop for this, though, can there be more than two levels? What we previously did was to first unpack references and then const/volatile. Can there be more? Meaning, is this really worth a loop? I certainly don't think there is a reasons to do it differently in different places.

(I'd prefer to keep the long comment here, though.)

Reply:
I've factored it out into a helper function.

I've kept the loop because I'm a bit paranoid about the order of ref/const/volatile and if that matters. I suspect it's unnecessary, but not completely sure.